configuration/reports/reportdescriptors/dataexports/sql/mentalhealthMedications.sql

```diff
@@ -0,0 +1,160 @@
+
+DROP TABLE IF EXISTS temp_obs;
+CREATE TEMPORARY TABLE temp_obs AS
+SELECT o.person_id, o.obs_id , o.obs_group_id , o.obs_datetime ,o.date_created , o.encounter_id, o.value_coded, o.concept_id, o.value_numeric , o.value_datetime , o.value_text , o.voided
+FROM temp_encounter te  INNER JOIN  obs o ON te.encounter_id=o.encounter_id
+WHERE o.voided =0;
+
+insert into temp_MH_meds(encounter_id, obs_group_id, patient_id, encounter_date)
+select
+e.encounter_id,
+obs_id,
+e.patient_id,
+e.encounter_datetime
+from temp_obs o
+inner join temp_encounter e on e.encounter_id  = o.encounter_id
+where concept_id = concept_from_mapping('PIH','10742'); -- prescription construct
+
+set @identifier_type ='0bc545e0-f401-11e4-b939-0800200c9a66';
+
+update temp_MH_meds t
+set emr_id = patient_identifier(t.patient_id, @identifier_type);
+
+update temp_MH_meds t
+set user_entered = encounter_creator(t.encounter_id);
+
+update temp_MH_meds t
+set provider = provider(t.encounter_id);
+
+update temp_MH_meds t
+set date_entered = encounter_date_created(t.encounter_id);
+
+update temp_MH_meds t
+set med_name = obs_from_group_id_value_coded_list(t.obs_group_id, 'PIH','10634','en');
+
+update temp_MH_meds t
+set dose_frequency = obs_from_group_id_value_coded_list(t.obs_group_id, 'PIH','9363','en');
+
+update temp_MH_meds t
+set dose_unit = obs_from_group_id_value_coded_list(t.obs_group_id, 'PIH','10744','en');
+
+update temp_MH_meds t
+set dose = obs_from_group_id_value_numeric(t.obs_group_id, 'PIH','9073');
+
+update temp_MH_meds t
+set duration = obs_from_group_id_value_numeric(t.obs_group_id, 'PIH','9075');
+
+update temp_MH_meds t
+set duration_unit = obs_from_group_id_value_coded_list(t.obs_group_id, 'PIH','6412', 'en');
+
+update temp_MH_meds t
+set route = obs_from_group_id_value_coded_list(t.obs_group_id, 'PIH','12651','en');
+
+update temp_MH_meds t
+set additional_medication_comments = obs_value_text(t.encounter_id, 'PIH','10637');
+
+-- The ascending/descending indexes are calculated ordering on the encounter date
+-- new temp tables are used to build them and then joined into the main temp table.
+drop temporary table if exists temp_MH_meds_index_asc;
+CREATE TEMPORARY TABLE temp_MH_meds_index_asc
+(
+    SELECT
+            patient_id,
+            encounter_date,
+            encounter_id,
+            index_asc
+FROM (SELECT
+            @r:= IF(@u = patient_id, @r + 1,1) index_asc,
+            encounter_date,
+            encounter_id,
+            patient_id,
+            @u:= patient_id
+      FROM temp_MH_meds,
+                    (SELECT @r:= 1) AS r,
+                    (SELECT @u:= 0) AS u
+            ORDER BY patient_id, encounter_date ASC, encounter_id ASC
+        ) index_ascending );
+
+CREATE INDEX tmia_e ON temp_MH_meds_index_asc(encounter_id);
+update temp_MH_meds tm
+inner join temp_MH_meds_index_asc tmia on tmia.encounter_id = tm.encounter_id
+set tm.index_asc = tmia.index_asc;
+
```

Comment:
What is the requirement for the index_asc/desc columns?  Is it supposed to order the meds for a patient?  The problem here is that the temp_MH_med_index_asc table is ordering all the meds within a patient, but then that is joined back into the temp_MH_meds table qualifying on encounter_id only, so you will get some weird results.  You will need something that uniquely identifies each row on temp_MH_meds to join.  You can introduce a sequential identifier on the that table and use that.  You would just use that it join the indexes in, you wouldn't need to select it in the final output.

Reply:
Thank you so much for your thorough review and the valuable feedback, along with clear instructions on addressing the concern. The purpose of the index_asc/desc columns is to order the medications within a patient. upon care observation, I noticed that the final output is producing unexpected results. Following your guidance, I took the necessary steps to resolve this issue. specifically, I introduced a unique identifier on the main table, utilizing it to accurately join the indexes with the main table; and that seems to produce the expected output.
@dmdesimone, @tommyiversonj

Reply:
Nice!... just one small thing.  I think you need to add meds_id to the ordering within the index generation.  Just so if rows have the same emr_id, encounter_date, and encounter_id the ascending and descending indexes will order them consistently.  

When you make that change you can go ahead and merge it in and someone will test it.